classes/class-wc-rest-connect-stripe-account-controller.php

```diff
@@ -21,12 +21,15 @@ public function get( $request ) {
 		$response = $this->stripe->get_account_details();
 
 		if ( is_wp_error( $response ) ) {
-			$response->add_data( array(
-				'message' => $response->get_error_message(),
-			), $response->get_error_code() );
-
 			$this->logger->debug( $response, __CLASS__ );
-			return $response;
+
+			return new WP_Error(
+				$response->get_error_code(),
+				$response->get_error_message(),
+				array(
+					'status' => 400
```

Comment:
Can you use the original HTTP code of the `$response`? AFAIK, the server won't ever return a `500` error unless something truly blew up.

How does Pressable behave with an `503` error? That's the closest code (semantically) to the "500". I'm thinking we should always use the status from the `$response` (except if it's 500, then replace it with something less offensive). Not only for the Stripe endpoints, but for everyone.

Reply:
Not without changes to the plugin's API layer. That $response is a WP_Error and doesn't include the HTTP response code. https://github.com/Automattic/woocommerce-services/blob/master/classes/class-wc-connect-api-client.php#L404

Reply:
Ouch. Something to look into when we improve user-facing errors in Q4. Go for it then, :shipit: